### PR TITLE
[FIX] l10n_cl: adapt report_invoice.xml to chilean-like format

### PR DESCRIPTION
**Pull request by Danisan**
The report in l10n_cl does not have a format as used in Chile

In this code, we added the necessary sections in order to have this report ready for inheriting it in electronic invoice implementation.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_cl/views/report_invoice.xml [hunk 1]
```diff
@@ -1,65 +1,247 @@
...
+                <span t-esc="o.partner_id.industry_id.name or ''"/>
+                <br/>
+            </div>
+            <div class="col-6">
+                <strong>Due Date:</strong>
+                <span t-esc="o.invoice_date_due" t-options='{"widget": "date"}'/>
+                <br/>
+                <strong>Address:</strong>
+                <span t-field="o.partner_id"
+                      t-options="{&quot;widget&quot;: &quot;contact&quot;, &quot;fields&quot;: [&quot;address&quot;], &quot;no_marker&quot;: true, &quot;no_tag_br&quot;: True}"/>
```
Comment (jco-odoo) [inline addons/l10n_cl/views/report_invoice.xml:99]: No need for &quot;  We can use regular ' or "  (should have been adapted for l10n_ar as well)

File: addons/l10n_cl/views/report_invoice.xml [hunk 2]
```diff
@@ -1,65 +1,247 @@
...
-            <attribute name="t-if">not o.l10n_latam_document_type_id.report_name</attribute>
-        </xpath>
+        <t t-set="o" position="after">
+            <t t-set="custom_header" t-value="'l10n_cl.custom_header'"/>
+            <t t-set="report_date" t-value="o.invoice_date"/>
+            <t t-set="report_number" t-value="int(o.l10n_latam_document_number)"/>
+            <t t-set="pre_printed_report" t-value="report_type == 'pdf'"/>
+            <t t-set="report_name" t-value="o.l10n_latam_document_type_id.name"/>
+            <t t-set="header_address" t-value="o.company_id.partner_id"/>
+            <t t-set="is_tax" t-value="o.l10n_latam_document_type_id.code not in ['39', '41', '110', '111', '112', '34']"/>
```
Comment (jco-odoo) [inline addons/l10n_cl/views/report_invoice.xml:131]: Should be set in the main report, not here.

File: addons/l10n_cl/views/report_invoice.xml [hunk 3]
```diff
@@ -1,65 +1,247 @@
...
             <attribute name="t-field">line.l10n_latam_price_unit</attribute>
         </xpath>
 
-        <xpath expr="//span[@id='line_tax_ids']" position="attributes">
-            <attribute name="t-esc">', '.join(map(lambda x: (x.description or x.name), line.l10n_latam_tax_ids))</attribute>
-        </xpath>
-
         <t t-set="current_subtotal" t-value="current_subtotal + line.price_subtotal" position="attributes">
-            <attribute name="t-value">current_subtotal + line.l10n_latam_price_subtotal</attribute>
+            <!--<t t-if="is_tax">-->
```
Comment (jco-odoo) [inline addons/l10n_cl/views/report_invoice.xml:147]: Need to check what we do with this...

File: addons/l10n_cl/views/report_invoice.xml [hunk 4]
```diff
@@ -1,65 +1,247 @@
...
+
+        <!-- custom header and footer -->
+        <t t-set="o" position="after">
+            <t t-set="custom_header" t-value="'l10n_cl.custom_header'"/>
+            <t t-set="report_date" t-value="o.invoice_date"/>
+            <t t-set="report_number" t-value="int(o.l10n_latam_document_number)"/>
+            <t t-set="pre_printed_report" t-value="report_type == 'pdf'"/>
+            <t t-set="report_name" t-value="o.l10n_latam_document_type_id.name"/>
+            <t t-set="header_address" t-value="o.company_id.partner_id"/>
+            <t t-set="is_tax" t-value="o.l10n_latam_document_type_id.code not in ['39', '41', '110', '111', '112', '34']"/>
```
Comment (jco-odoo) [inline addons/l10n_cl/views/report_invoice.xml:217]: We can set this in the report itself.  No need to do that here as we pass parameters.

File: addons/l10n_cl/views/report_invoice.xml (hunk 1 above)
Comment (jco-odoo) [reply addons/l10n_cl/views/report_invoice.xml:99]: TODO

File: addons/l10n_cl/views/report_invoice.xml (hunk 2 above)
Comment (jco-odoo) [reply addons/l10n_cl/views/report_invoice.xml:131]: Can you check

File: requirements.txt [hunk 5]
```diff
@@ -21,6 +21,7 @@ mock==2.0.0
 num2words==0.5.6
 ofxparse==0.19
 passlib==1.7.1
+pdf417gen==0.7.1
```
Comment (jco-odoo) [inline requirements.txt:24]: so this line needs to be removed and I would need @d-fence to add it in the runbot image + document it in the l10n_cl_edi manifest.

**Review by jco-odoo — CHANGES_REQUESTED** [review]
@Danisan @suv-bmya Can you take care of this please.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -12,3 +12,5 @@ TL;DR
   when reporting issues. Please search for duplicates first!
 * Pull requests must be made against the [correct version](https://github.com/odoo/odoo/wiki/Contributing#against-which-version-should-i-submit-a-patch)
 * There are restrictions on the kind of [changes allowed in stable series](https://github.com/odoo/odoo/wiki/Contributing#what-does-stable-mean)
+
```
[the remark this answers is not in the document]
Comment (Danisan) [reply CONTRIBUTING.md:15]: fingers..

File: addons/l10n_cl/views/report_invoice.xml (hunk 2 above)
Comment (Danisan) [reply addons/l10n_cl/views/report_invoice.xml:131]: in the custom_header you mean?

File: addons/l10n_cl/views/report_invoice.xml (hunk 3 above)
Comment (Danisan) [reply addons/l10n_cl/views/report_invoice.xml:147]: by now.. do I remove the comment lines?

File: requirements.txt (hunk 5 above)
Comment (Danisan) [reply requirements.txt:24]: there is a bullet added in the description to l10n_cl_edi, explaining this.

File: addons/l10n_cl/views/report_invoice.xml (hunk 2 above)
Comment (Danisan) [reply addons/l10n_cl/views/report_invoice.xml]: @jco-odoo moved to the custom_header: 4d1f10a546e5de417bae9b2a12bd6bc3de92418c

File: addons/l10n_cl/views/report_invoice.xml (hunk 4 above)
Comment (Danisan) [reply addons/l10n_cl/views/report_invoice.xml]: Where do you mean?

File: requirements.txt (hunk 5 above)
Comment (Danisan) [reply requirements.txt:24]: ok removed, but why don't define it in requirements?

File: addons/l10n_cl/views/report_invoice.xml (hunk 1 above)
Comment (Danisan) [reply addons/l10n_cl/views/report_invoice.xml]: solved

File: addons/l10n_cl/views/report_invoice.xml (hunk 4 above)
Comment (Danisan) [reply addons/l10n_cl/views/report_invoice.xml]: ready

File: addons/l10n_cl/views/report_invoice.xml (hunk 3 above)
Comment (Danisan) [reply addons/l10n_cl/views/report_invoice.xml]: talked about this in previous session 2020-09-08 it is ok (removed comments)

File: addons/l10n_cl/views/res_config_settings_view.xml [hunk 7]
```diff
@@ -10,11 +9,11 @@
                 <div id="l10n_cl_title" attrs="{'invisible': True}">
                     <h2>Chilean Localization</h2>
                 </div>
-                <div id="l10n_cl_section" class="row mt16 o_settings_container">
+                <div id="l10n_cl_section" class="row mt16 o_settings_container" groups="account.group_account_manager">
                     <!-- inside empty to add configuration of tags -->
+                    <div class="col-12 col-lg-6 o_setting_box" id="l10n_cl_edi_install"/>
```
[the remark this answers is not in the document]
Comment (Danisan) [reply addons/l10n_cl/views/res_config_settings_view.xml:14]: I missed this.. sorry

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**